packages/ui/react-ui-table/src/util/table-controls.ts

```diff
@@ -2,26 +2,89 @@
 // Copyright 2024 DXOS.org
 //
 
-export const checkboxAttributes = {
+// Core types.
+export type TableControl = 'checkbox' | 'switch';
+
+export type ControlData =
+  | { type: 'checkbox'; rowIndex: number; header?: boolean }
+  | { type: 'switch'; colIndex: number; rowIndex: number };
+
+type CommonRenderProps = {
+  checked?: boolean;
+  disabled?: boolean;
+};
+
+export type RenderCheckboxProps = CommonRenderProps & Omit<Extract<ControlData, { type: 'checkbox' }>, 'type'>;
+export type RenderSwitchProps = CommonRenderProps & Omit<Extract<ControlData, { type: 'switch' }>, 'type'>;
+
+export const CONTROL_IDENTIFIERS = {
   checkbox: 'data-table-checkbox',
-  header: 'data-table-header',
+  switch: 'data-table-switch',
 } as const;
 
-export type RenderCheckboxProps = { rowIndex: number; header?: boolean; checked?: boolean; disabled?: boolean };
+const BASE_CLASSES = {
+  checkbox: 'absolute inset-block-[6px] inline-end-[6px] ch-checkbox',
+  switch: 'absolute inset-block-[4px] inline-end-[4px] ch-checkbox--switch',
```

Comment:
Gorgeous ✨

Recommend using rem for these tho:
```suggestion
  checkbox: 'absolute inset-block-[.375rem] inline-end-[.375rem] ch-checkbox',
  switch: 'absolute inset-block-[.25rem] inline-end-[.25rem] ch-checkbox--switch',
```